projects/step-core/src/lib/modules/table/shared/table-remote-data-source.ts

```diff
@@ -107,7 +110,8 @@ export class TableRemoteDataSource<T> implements TableDataSource<T> {
   constructor(
     private _tableId: string,
     private _rest: TableRestService,
-    private _requestColumnsMap: { [key: string]: string }
+    private _requestColumnsMap: { [key: string]: string },
+    private _filter?: string
```

Comment:
I've just checked and noticed that some tables have this filter as dynamic value.
For example table in the `searchResourceDialog.html`

I would suggest to add filter a input parameter to `table.component`. And link this parameter with the subject
```
private _filter$ = new BehaviorSubject<string>(undefined);

get filter(): string {
  return this._filter$.value;
}

@Input() set filter(value: string) {
   if (value === this.filter) {
     return;
   }
  this._filter$.next(value);
}

```

Then add this subject to the combineLatest with page$,  sort$ and search$
```
combineLatest([page$, sort$, this._search$, this._filter$])
      .pipe(takeUntil(this._terminator$))
      .subscribe(([page, sort, search, filter]) => tableDataSource.getTableData(page, sort, search, filter));
```

And as final step extend `getTableData` method signature with new parameter.
For TableLocalDataSource the filter parameter will be ignored

Reply:
Done! Yes indeed much more convenient and dynamic and the old `st-table` directive had the same feature as well

Reply:
@chillbirdo Thank you. Looks nice!
Just a few tiny improvements:

- I guess that _filter argument can be removed from TableRemoteDataSource constructor. I don't see that it's used anymore
- Also in `table.component.ts` can you complete the `_filter$` subject in the `ngOnDestroy` method as it is done for `_search$`

Reply:
Yes, subjects need to be completed, I overlooked that, thanks! I'll check the `_filter` argument too

Reply:
done!